run-automation-test.md

```diff
@@ -0,0 +1,143 @@
+ + For instruction on how to install and configure TeamCity server, follow [this guide](https://confluence.jetbrains.com/display/TCD18/Installing+and+Configuring+the+TeamCity+Server).
+
+Set up your project and link your app repository that we will test with Kobiton later.
++ Instruction on how to integrate GitHub with TeamCity can be found [here](https://confluence.jetbrains.com/display/TCD10/Integrating+TeamCity+with+VCS+Hosting+Services).
+ 
+>For more information about TeamCity can be found in [this tutorial](https://confluence.jetbrains.com/display/TCD10/TeamCity+Documentation).
+
+## 2. Configure automation test script
+### 2.1 Get Username and API key
+Go to https://portal.kobiton.com and login to your Kobiton account.
+- Username
+  + Click *user icon* -> **"Profile"** (you might find it in the top right corner)
+
+  ![get username](./assets/username.png )
+
+- API key 
+  + Click *user icon* -> **"Settings"**
+
+  ![get api key](./assets/api_key.png )
+
+### 2.2 Get desired capabilities
+The desired capabilities need to be added to the automation test script to allow the test to be executed on Kobiton device.
+  + Click **Devices** in the top navigation bar.
+ 
+    ![click devices](./assets/devices.png )
+ 
+  + Hover over any device you want to test with and click on the Automation settings button (the gear symbol).
+ 
+    ![click auto](./assets/click_auto.png )
+ 
+  + On the left-hand side, you can select your preferred language, App Type, etc.
+  In this example, we use **NodeJS** as the default language of the script. Therefore, choose **NodeJS** in `Language` section and **Hybrid/Native from Apps** in `App type` section.
+
+    ![click auto](./assets/automation.png ) 
+### 2.3 Configure automation test script
+Kobiton has already provided sample scripts for automation testing, visit [here](https://github.com/kobiton/samples) for reference. 
+> Note: In this guideline, we will use the Node.js sample (samples/javascript folder) as an example.
+
+Open automation test script file in your repository or create a new one.
+
+Add necessary variables as well as `kobitonServerConfig` and `desiredCaps` in the script with ones collected in the previous step.
+
+Example:
+
+```javascript
+const username = '<YOUR_KOBITON_USERNAME>'
+const apiKey = '<YOUR_KOBITON_API_KEY>'
+const kobitonServerConfig = {
+  protocol: 'https',
+  host: 'api-test.kobiton.com',
+  auth: `${username}:${apiKey}`
+}
+var desiredCaps = {
+  sessionName:        'Automation test session',
+  sessionDescription: '', 
+  deviceOrientation:  'portrait',  
+  captureScreenshots: true, 
+  app:                '<APP_URL>', 
+  deviceGroup:        'KOBITON', 
+  deviceName:         '<DEVICE_NAME>',
+  platformVersion:    '<DEVICE_VERSION>',
+  platformName:       '<DEVICE_PLATFORM_NAME>' 
+}
+```
+### 2.4 Configure TeamCity project
```

Comment:
Assumption for this section is whether reader already had the knowledge about TeamCity or not?
If not --> this section is too general, as User (without the knowledge about TeamCity), I cannot understand the process --> require more explanations for TeamCity:
- What is TeamCity?
- Why we need TeamCity? (different with traditional way for automation test)
- What TeamCity can do?
- ...